.circleci/config.yml

```diff
@@ -19,21 +19,51 @@ commands:
   check-uncommitted:
     description: Check for uncommitted changes.
     steps:
-      - run: 
-          name: Check for uncomitted changes using git porcelain command.
+      - run:
+          name: Check for uncomitted changes.
           command: |
             if [[ `git status --porcelain` ]]; then
               echo "ERROR: uncommitted changes"
               git status --porcelain
               exit 1
             fi
+  install-golangci-lint:
+    description: Install golangci-lint
+    parameters:
+      version:
+        type: string
+        default: 1.21.0
+      gobin:
+        type: string
+        default: /go/bin
+      prefix:
+        type: string
+        default: v1
+        description: Prefix for cache key to store the binary.
+    steps:
+      - restore_cache:
+          name: Restore golangci-lint cache
+          keys: ['<< parameters.prefix >>-golangci-lint-{{ arch }}-<< parameters.version >>']
+      - run:
+          name: Install golangci-lint
+          command: |
+            mkdir -p << parameters.gobin >>
+            command -v << parameters.gobin >>/golangci-lint && exit
+
+            download=https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh
+            wget -O- -q $download | sh -s -- -b << parameters.gobin >>/ v<< parameters.version >>
```

Comment:
Nit: The server uses `curl` instead (https://github.com/mattermost/mattermost-server/blob/master/.circleci/config.yml#L67). Maybe we want to sync this?

Reply:
I'd be inclined to just leave as-is -- this is copied and pasted from an existing orb, and I think it's probably fine to diverge.